### PR TITLE
Attached sampling

### PR DESCRIPTION
**Pull request by ChengshuLi**

### CODE REVIEW

File: omnigibson/utils/bddl_utils.py [hunk 1]
```diff
@@ -191,7 +191,7 @@ def process_single_condition(condition):
     "insource": ObjectStateInsourcePredicate,
 }
 
-KINEMATIC_STATES_BDDL = frozenset([state.__name__.lower() for state in _KINEMATIC_STATE_SET])
+KINEMATIC_STATES_BDDL = frozenset([state.__name__.lower() for state in _KINEMATIC_STATE_SET] + ["attached"])
```
Comment (cremebrule) [inline omnigibson/utils/bddl_utils.py:194]: why is AttachedTo not a kinematic state?

File: omnigibson/utils/bddl_utils.py [hunk 2]
```diff
@@ -846,6 +897,10 @@ def _filter_object_scope(self, input_object_scope, conditions, condition_type):
                             # Reset if we're sampling a kinematic state
                             if condition.STATE_NAME in {"inside", "ontop", "under"}:
                                 kwargs["reset_before_sampling"] = True
+                            elif condition.STATE_NAME in {"attached"}:
+                                kwargs["bypass_alignment_checking"] = True
+                                kwargs["check_physics_stability"] = True
+                                kwargs["can_joint_break"] = False
```
Comment (cremebrule) [inline omnigibson/utils/bddl_utils.py:903]: add TODO here to note that we need to revert this when possible in the future?

File: omnigibson/utils/bddl_utils.py [hunk 3]
```diff
@@ -965,53 +1077,53 @@ def _import_sampleable_objects(self):
...
+                model_choices = set()
+                for category in categories:
+                    # Get all available models that support all of its synset abilities
+                    model_choices = set(get_all_object_category_models_with_abilities(
                         category=category,
-                        model=model,
-                        prim_type=PrimType.CLOTH if "cloth" in OBJECT_TAXONOMY.get_abilities(obj_synset) else PrimType.RIGID,
-                    )
-                    num_new_obj += 1
+                        abilities=OBJECT_TAXONOMY.get_abilities(OBJECT_TAXONOMY.get_synset_from_category(category)),
```
Comment (cremebrule) [inline omnigibson/utils/bddl_utils.py:1095]: This code seems to change a lot from the previous iteration. Are we sure this doesn't break sampling?

**Review by cremebrule — APPROVED** [review]
overall LGTM, some clarifying comments

File: omnigibson/utils/bddl_utils.py (hunk 1 above)
Comment (ChengshuLi) [reply omnigibson/utils/bddl_utils.py:194]: Not sure, but I don't want to change the current behavior. We can make it inherit KinematicStateMixin later.

File: omnigibson/utils/bddl_utils.py (hunk 2 above)
Comment (ChengshuLi) [reply omnigibson/utils/bddl_utils.py:903]: why do we need to revert this?

File: omnigibson/utils/bddl_utils.py (hunk 3 above)
Comment (ChengshuLi) [reply omnigibson/utils/bddl_utils.py:1095]: most of them are indentation changes. Yeah I tested some other task and I think it doesn't break.

**Pull request merged**